### PR TITLE
CNV-46603: add CUDN kind

### PR DESCRIPTION
**Pull request by jpinsonneau**
Followup on https://github.com/openshift/networking-console-plugin/pull/134 to add `ClusterUserDefinedNetwork` to list / forms

image
image
image

### CODE REVIEW

File: src/utils/components/MatchExpression/Expression.tsx [hunk 1]
```diff
@@ -0,0 +1,111 @@
...
+              isFullWidth
+              onClick={() => setIsDropdownOpen(!isDropdownOpen)}
+              ref={toggleRef}
+            >
+              {expression.operator ? <>{expression.operator}</> : t('Select an operator')}
+            </MenuToggle>
+          )}
+        >
+          <DropdownList>
+            {allowedOperators.map((o, i) => (
```
Comment (upalatucci) [inline src/utils/components/MatchExpression/Expression.tsx:68]:
```suggestion
            {allowedOperators.map((operator, i) => (
```

Can we please use a more explanatory name in this map?

File: src/utils/components/MatchExpression/Expression.tsx [hunk 2]
```diff
@@ -0,0 +1,111 @@
+import * as React from 'react';
```
Comment (upalatucci) [inline src/utils/components/MatchExpression/Expression.tsx:1]:
```suggestion
import React, { FC } from 'react';
```
Please remove all the `React.` usage and import react like this.


Console use to import react like this for a typescirpt issue that we don't have in this repo.

File: src/views/udns/form/components/CUDNNamespaceSelector.tsx [hunk 3]
```diff
@@ -0,0 +1,77 @@
...
+          <FormGroup
+            fieldId="basic-settings-matchExpressions"
+            label={
+              <Popover
+                bodyContent={
+                  <Flex direction={{ default: 'column' }}>
+                    <FlexItem>{` - ${t('key is the label key that the selector applies')}`}</FlexItem>
+                    <FlexItem>{` - ${t("operator represents a key's relationship to a set of values. Valid operators are In, NotIn, Exists and DoesNotExist.")}`}</FlexItem>
+                    <FlexItem>{` - ${t('values is an array of string values. If the operator is In or NotIn, the values array must be non-empty. If the operator is Exists or DoesNotExist, the values array must be empty. This array is replaced during a strategic merge patch.')}`}</FlexItem>
+                  </Flex>
```
Comment (upalatucci) [inline src/views/udns/form/components/CUDNNamespaceSelector.tsx:57]: Can this be a List? https://v5-archive.patternfly.org/components/list/html/#ws-page-title

File: src/views/udns/form/components/UDNLayer2Parameters.tsx [hunk 4]
```diff
@@ -39,14 +63,51 @@ const Layer2Parameters: FC = () => {
...
+                  {` - ${t(
+                    'Primary network is automatically assigned to every pod created in the same namespace.',
+                  )}`}
+                </FlexItem>
+                <FlexItem>
+                  {` - ${t(
+                    'Secondary network is only assigned to pods that use k8s.v1.cni.cncf.io/networks annotation to select given network.',
+                  )}`}
+                </FlexItem>
+              </Flex>
```
Comment (upalatucci) [inline src/views/udns/form/components/UDNLayer2Parameters.tsx:104]: And here also [list](https://v5-archive.patternfly.org/components/list/html/#ws-page-title)

File: src/views/udns/form/components/UDNSubnets.tsx [hunk 5]
```diff
@@ -17,10 +17,10 @@ import {
 } from '@utils/resources/udns/types';
 
 type UDNSubnetsProps = {
-  description: string;
+  description: React.ReactNode;
   onSubnetsChange: (subnets: UserDefinedNetworkSubnet[]) => void;
   subnets: UserDefinedNetworkSubnet[];
-  title: string;
+  title: React.ReactNode;
```
Comment (upalatucci) [inline src/views/udns/form/components/UDNSubnets.tsx:23]: import react like 

`import React, { ReactNode } from 'react'`

File: src/utils/components/MatchExpression/Expression.tsx (hunk 1 above)
Comment (jpinsonneau) [reply src/utils/components/MatchExpression/Expression.tsx:68]: `operator` is already declared above so I used `allowedOperator` instead

**Comment by jpinsonneau** [comment]
Rebased and addressed feedback

**Pull request merged**